mm2src/coins/lp_coins.rs

```diff
@@ -3570,27 +3590,149 @@ impl MmCoinStruct {
+        // calc dex fee
+        let rate = Self::dex_fee_rate(taker_coin.ticker(), rel_ticker);
+        let dex_fee = trade_amount * &rate;
+        let min_tx_amount = MmNumber::from(taker_coin.min_tx_amount());
+        if dex_fee <= min_tx_amount {
+            return DexFee::Standard(min_tx_amount);
+        }
+
+        if taker_coin.should_burn_dex_fee() {
+            if taker_coin.is_kmd() {
+                // use a special dex fee option for kmd
+                let (fee_amount, burn_amount) = Self::calc_burn_amount_for_op_return(&dex_fee, &min_tx_amount);
+                return DexFee::WithBurn {
+                    fee_amount,
+                    burn_amount,
+                    burn_destination: DexFeeBurnDestination::KmdOpReturn,
+                };
+            } else {
+                // burn dex fee to the burn account
+                let (fee_amount, burn_amount) = Self::calc_burn_amount_for_burn_account(&dex_fee, &min_tx_amount);
+                // burn_amount can be set to zero if it is dust
+                if burn_amount > MmNumber::from(0) {
+                    return DexFee::WithBurn {
+                        fee_amount,
+                        burn_amount,
+                        burn_destination: DexFeeBurnDestination::BurnAccount,
+                    };
+                }
+            }
         }
+        DexFee::Standard(dex_fee)
+    }
+
+    /// Returns dex fee discount if KMD is traded
+    pub fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
+        let fee_discount_tickers: &[&str] = if var("MYCOIN_FEE_DISCOUNT").is_ok() {
+            &["KMD", "MYCOIN"] // used in tests
+        } else {
+            &["KMD"]
+        };
+        if fee_discount_tickers.contains(&base) || fee_discount_tickers.contains(&rel) {
+            // 1/777 - 10%
+            BigRational::new(9.into(), 7770.into()).into()
+        } else {
+            BigRational::new(1.into(), 777.into()).into()
+        }
+    }
+
+    fn calc_burn_amount_for_op_return(dex_fee: &MmNumber, min_tx_amount: &MmNumber) -> (MmNumber, MmNumber) {
+        // Drop the dex fee by 25%, which will be burned during the taker fee payment.
+        //
+        // This cut will be dropped before return if the final amount is less than
+        // the minimum transaction amount.
+
+        // Dex fee with 25% cut
+        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);
+
+        if &new_fee >= min_tx_amount {
+            // Use the max burn value, which is 25%.
+            let burn_amount = dex_fee - &new_fee;
+
+            // we don't care if burn_amount < dust as any amount can be sent to op_return
+            (new_fee, burn_amount)
+        } else {
+            // Burn only the exceed amount because fee after 25% cut is less
+            // than `min_tx_amount`.
+            let burn_amount = dex_fee - min_tx_amount;
+
+            (min_tx_amount.clone(), burn_amount)
+        }
```

Comment:
could you remove please empty lines

```rust
        // Dex fee with 25% cut
        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);
        if &new_fee >= min_tx_amount {
            // Use the max burn value, which is 25%.
            let burn_amount = dex_fee - &new_fee;
            // we don't care if burn_amount < dust as any amount can be sent to op_return
            (new_fee, burn_amount)
        } else {
            // Burn only the exceed amount because fee after 25% cut is less than `min_tx_amount`.
            let burn_amount = dex_fee - min_tx_amount;
            (min_tx_amount.clone(), burn_amount)
        }
```